lib/db_adapter.go

```diff
@@ -10,6 +10,14 @@ type DbAdapter struct {
 	snapshot   *Snapshot
 }
 
+func NewDbAdapter(chain *Blockchain) *DbAdapter {
```

Comment:
To match below, maybe this makes more sense w/ a signature like `func (chain *Blockchain) GetDbAdapter *DbAdapter {`.

Reply:
agreed with mf ^